File: RiderRadio/CONFIGURATION/Tools/Parallax/Parallax.m
//
//  Parallax.m
//
//  Created by Fabien on 02/09/2014.
//  Copyright (c) 2014 Fabien Moussavi. All rights reserved.
//


#import <Foundation/Foundation.h>
#import "Parallax.h"


@implementation Parallax

/*
 * To Have a paralax effect on the view passed in argument
 */
+ (void)registerEffectForView:(UIView *)view withDepth:(CGFloat)depth
{
    UIInterpolatingMotionEffect *effectX = [[UIInterpolatingMotionEffect alloc] initWithKeyPath:@"center.x" type:UIInterpolatingMotionEffectTypeTiltAlongHorizontalAxis];
    UIInterpolatingMotionEffect *effectY = [[UIInterpolatingMotionEffect alloc] initWithKeyPath:@"center.y" type:UIInterpolatingMotionEffectTypeTiltAlongVerticalAxis];
    
    [effectX setMinimumRelativeValue:@(depth)];
    [effectX setMaximumRelativeValue:@(-depth)];
    [effectY setMinimumRelativeValue:@(depth)];
    [effectY setMaximumRelativeValue:@(-depth)];
    
    UIMotionEffectGroup *group = [[UIMotionEffectGroup alloc] init];
    group.motionEffects = @[effectX, effectY];
    
    [view addMotionEffect:group];
}

@end

File: RiderRadio/CONFIGURATION/Tools/Parallax/Parallax.swift
//
//  Parallax.swift
//
//  Created by Fabien on 31/07/2014.
//  Copyright (c) 2014 Fabien Moussavi. All rights reserved.
//


import UIKit


//===========//
/*   CLASS   */
//===========//
class Parallax {
    
    //**/ Statics
    //====================================================================================================//
    /*
    * To Have a paralax effect on the view passed in argument
    */
    class func registerEffectForView(aView: UIView, depth: CGFloat)
    {
        var effectX: UIInterpolatingMotionEffect = UIInterpolatingMotionEffect(keyPath: "center.x", type: UIInterpolatingMotionEffectType.TiltAlongHorizontalAxis)
        var effectY: UIInterpolatingMotionEffect = UIInterpolatingMotionEffect(keyPath: "center.y", type: UIInterpolatingMotionEffectType.TiltAlongVerticalAxis)
        
        effectX.minimumRelativeValue = -depth
        effectX.maximumRelativeValue = depth
        effectY.minimumRelativeValue = -depth
        effectY.maximumRelativeValue = depth
        
        var group: UIMotionEffectGroup = UIMotionEffectGroup()
        group.motionEffects = [effectX, effectY]
        
        aView.addMotionEffect(group)
    }
    
}

